compiler/rustc_trait_selection/src/solve/mod.rs

```diff
@@ -437,18 +428,18 @@ impl<'tcx> EvalCtxt<'_, 'tcx> {
 }
 
 #[instrument(level = "debug", skip(infcx), ret)]
-fn take_external_constraints<'tcx>(
+fn take_and_intern_external_constraints<'tcx>(
```

Comment:
i feel like the `intern` isn't too useful here as the exact way external constraints are stored is irrelevant for the caller :thinking: 

would keep the current name (or actually rename it to `compute_external_query_constraints` or sth)